langkit/templates/c_api/iterator_types_c.mako

```diff
@@ -0,0 +1,27 @@
+## vim: filetype=makocpp
+
+<%def name="incomplete_decl(cls)">
+<% type_name = cls.c_type(capi).name %>
+${c_doc(cls)}
+typedef void* ${type_name};
+</%def>
+
+<%def name="decl(cls)">
+
+<% type_name = cls.c_type(capi).name %>
+
+/* Set the next value of the iterator in the given element pointer.
+   Return 1 if successful, otherwise 0.  */
```

Comment:
These docs are supposed to be shared in `langkit.documentation`: could you move them and use `${c_doc()}` to replace them where needed?